apps/client/src/routes/dashboards/dashboards-index/hooks/use-dashboards-query.ts

```diff
@@ -0,0 +1,26 @@
+import { useQuery } from '@tanstack/react-query';
+import { useIntl } from 'react-intl';
+
+import { DASHBOARDS_QUERY } from '~/data/dashboards';
+import { useSendNotification } from '~/hooks/notifications/use-send-notification';
+import { ApiError } from '~/services';
+
+export function useDashboardsQuery() {
+  const intl = useIntl();
+  const sendNotification = useSendNotification();
+
+  return useQuery({
+    ...DASHBOARDS_QUERY,
+    onError: (error) => {
+      if (error instanceof ApiError) {
```

Comment:
Each query handles its own error then, might be worth having a helper for this.

Reply:
Definitely. For notifications in general, we will likely have repeated patterns that would benefit from helpers.

Reply:
Leaving this refactor out of this PR to instead include as part of an upcoming change to improve error boundaries.